### PR TITLE
New package: png++-0.2.9

### PR DESCRIPTION
**Pull request by misuchiru03**

### CODE REVIEW

**Comment by maxice8** [comment]
package should be called png++ and should have a png++-devel subpackage

**Comment by abenson** [comment]
png++ would be empty, though

**Comment by maxice8** [comment]
the 3 reset commits are not meant to be there

File: srcpkgs/png++-devel/template [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'png++-devel'
+pkgname=png++-devel
```
Comment (maxice8) [inline srcpkgs/png++-devel/template]: should be png++

File: srcpkgs/png++-devel/template [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# Template file for 'png++-devel'
```
Comment (maxice8) [inline srcpkgs/png++-devel/template]: same for here

File: srcpkgs/png++-devel/template [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+pkgname=png++-devel
+version=0.2.9
+revision=1
+noarch=yes
+_pkgname=png++
```
Comment (maxice8) [inline srcpkgs/png++-devel/template]: no need for this

File: srcpkgs/png++-devel/template [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+build_style=gnu-makefile
+make_build_target="docs"
```
Comment (maxice8) [inline srcpkgs/png++-devel/template]: This **might** make it not build the main target, make sure it does build the main target

Comment (abenson) [reply srcpkgs/png++-devel/template]: The main target is an example and is not needed nor packaged.

File: srcpkgs/png++-devel/template (hunk 1 above)
Comment (abenson) [reply srcpkgs/png++-devel/template]: So no -devel package, headers in the main package

File: srcpkgs/Brainfunk/template [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+build_style=gnu-makefile
+makedepends="png++-devel gcc make"
+depends="gcc"
+short_desc="Brainfuck optimizing interpreter and compiler"
+maintainer="misuchiru03 <misuchiru03+void@gmail.com>"
+license="GPL-3.0-or-later"
+homepage="https://github.com/GReaperEx/Brainfunk"
+distfiles="${homepage}/archive/v${version}-beta.tar.gz"
+checksum=6b48bd1fbe0dbce20f3e8af5c799023ea820ddc45aca8fac0ccecfa2de359ee2
+do_install() {
```
Comment (abenson) [inline srcpkgs/Brainfunk/template]: blank line between checksum and do_install

File: srcpkgs/png++-devel/template [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+license="BSD-3-Clause"
+homepage="https://www.nongnu.org/pngpp"
+distfiles="${NONGNU_SITE}/pngpp/${_pkgname}-${version}.tar.gz"
+checksum=abbc6a0565122b6c402d61743451830b4faee6ece454601c5711e1c1b4238791
+
+post_install() {
+	vlicense COPYING
+}
+
+png++-devel-doc_package() {
```
Comment (abenson) [inline srcpkgs/png++-devel/template]: After some thought, this should be `png++-docs`. Update the symlink as well

**Comment by abenson** [comment]
For some background, `png++` is a "header only" library, and either the headers (the only files in it) go in `png++` or `png++-devel`. It's not possible to split `png++` and `png++-devel`, unless `png++` is a meta package.

**Comment by maxice8** [comment]
so have only png++ like we have json-c++

**Pull request merged**